parlai/mturk/tasks/acute_eval/fast_eval.py

```diff
@@ -0,0 +1,626 @@
+        self.config_ids: List[str] = list(ids)
+        self.config_ids.sort()
+        self.combos: List[Tuple[str, str]] = list(combos)
+        self.combos.sort()
+        # verify that ids are contained in the config:
+        for config_id in self.config_ids:
+            if config_id not in choices:
+                raise RuntimeError(
+                    f'ID {config_id} not specified in the config (`configs.py`).'
+                )
+        assert len(self.config_ids) > 1, 'Must specify least 2 ids'
+
+    def _print_progress(self, msg: str):
+        """
+        Format a msg to print to stdout well.
+
+        :param msg:
+            message to print
+        """
+        print(f"\n{'-' * 60}\n {msg} \n {'-' * 60}")
+
+    def _get_selfchat_config(self, config_id: str) -> Dict[str, Any]:
+        """
+        Return config for selfchat.
+
+        :param config_id:
+            config_id string
+
+        :return config:
+            dict config for self-chat
+        """
+        outfile = self._get_selfchat_log_path(config_id)
+        config = CONFIG[config_id]
+        config.update(
+            {
+                'task': self.task,
+                'outfile': outfile,
+                'num_self_chats': NUM_SELFCHAT_EXAMPLES,
+                'selfchat_max_turns': SELFCHAT_MAX_TURNS,
+                'display_examples': False,
+                'log_every_n_secs': -1,
+                'indent': -1,
+            }
+        )
+        return config
+
+    def _get_task_conversion_config(self, config_id: str) -> Dict[str, Any]:
+        """
+        Return config for task conversion to conversation format.
+
+        :param config_id:
+            config_id string
+
+        :return config:
+            dict config for task conversion script
+        """
+        outfile = self._get_task_data_path(config_id)
+        config = CONFIG[config_id]
+        config.update(
+            {
+                'outfile': outfile,
+                'num_episodes': NUM_TASK_DATA_EPISODES,
+                'speaker_0_id': f'{config_id}_as_human',
+                'speaker_1_id': config_id,
+            }
+        )
+        return config
+
+    def _get_vs_path(self, subdir: str) -> str:
+        """
+        Return a unique path for the set of comparison combos given a subdirectory.
+
+        We hash the filename as it can grow quite large.
+        """
+        assert subdir
+        os.makedirs(os.path.join(self.root_dir, subdir), exist_ok=True)
+
+        def _combo_name(id1, id2):
+            """
+            Return joined name for combo of comparisons.
+            """
+            id1_name = id1
+            id2_name = id2
+            if 'model' in CONFIG[id1]:
+                id1_name += self.task.replace(':', '_')
+            if 'model' in CONFIG[id2]:
+                id2_name += self.task.replace(':', '_')
+            return f'{id1_name}__vs__{id2_name}'
+
+        return os.path.join(
```

Comment:
Personally I find it more helpful to hash date + model_pairs instead of just model_pairs. One use case, if one runs the acute-eval on the model_pair lists and  accidentally responds with 'o' to overwrite pairing files during the second/third/... run, the pairing file of the first run can never be retrieved, which might be of great help for debugging the acute-eval logs.

Reply:
not a bad idea, i'll add that in

Reply:
actually, i think i'd rather not touch that, since then it would make a new pairings file every day